File: Beginner/exEncode/exEncode.java
//run as java -ea exEncode
import java.util.Scanner;

public class exEncode{
	public static void main(String[] args){
		int T,N,x;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100) : "T must be 1 <= T <= 100";

		for(int t=1;t<=T;t++){
			N = scan.nextInt();
			assert(N>=1 && N<=10000) : "N must be 1 <= N <= 10^4";

			int[] A = new int[N];
			int[] B = new int[N];

			for(int i=0;i<N;i++){
				x = scan.nextInt();
				A[i] = (x<<16)>>16;
				B[i] = (x-A[i])>>16;
			}

			System.out.println("Case "+ t + ":");

			for(int i=0;i<N;i++){
				System.out.print(A[i] + " ");
			}

			System.out.println();

			for(int i=0;i<N;i++){
				System.out.print(B[i] + " ");
			}

			System.out.println();
		}
	}
}

File: Beginner/exEncode/exEncode.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,N,x;

	scanf("%d",&T);
	assert(T>=1 && T<=100);

	for(int t=1;t<=T;t++){
		scanf("%d",&N);
		assert(N>=1 && N<=10000);

		int A[N];
		int B[N];

		for(int i=0;i<N;i++){
			scanf("%d",&x);
			A[i] = (x<<16)>>16;
			B[i] = (x-A[i])>>16;
		}

		printf("Case %d:\n",t);

		for(int i=0;i<N;i++){
			printf("%d ",A[i]);
		}

		printf("\n");

		for(int i=0;i<N;i++){
			printf("%d ",B[i]);
		}

		printf("\n");
	}
	
	return 0;
}